### PR TITLE
Remove old edx-status-bot messages from PR's

### PR DESCRIPTION
**Pull request by michaelyoungstrom**
This bot is causing too much clutter on PR's. Deleting old messages before posting a new one alleviates this clutter and will hopefully get more people using this feature.

### CODE REVIEW

**Comment by michaelyoungstrom** [comment]
@UmanShahzad I think your review makes sense. If we're going to use this marker/action model, we should be consistent. Though this means:
1- old comments will be deleted as soon as this gets run again, even if we aren't yet posting a new comment (this is fine with me, because the fact that its running shows that any old message is now obsolete).
2- it feels weird to have the remove comments marker just return True...

I'd like to see what @jmbowman thinks of this

**Comment by UmanShahzad** [comment]
@michaelyoungstrom Yeah I don't think the action/marker model is perfect for all use cases -- none would be -- but it does help in seeing how the code flows more easily.

And the fact that it just returns `True` is an implementation detail that fits the use case as I see it just fine; if we wanted to always check whether a comment existed first before deleting, we could write that logic but it'd be redundant since we need to do that check anyway to actually delete; if we wanted to check for a flag or any other kind of marker we could do that.

**Comment by michaelyoungstrom** [comment]
@jmbowman That shouldn't happen. The only weird scenario I can think of is when there's problems with the deduper after the spigot is released (all the contexts show failures from an old build being aborted, then the real job results come in one by one). But even in this scenario, each time a new context comes in, old comments will be deleted, and since there are no pending contexts left, a new comment will be posted.

So, basically, even if this got triggered after tests finished, a new comment would be posted, so the deletion doesn't bring any harm IMO.
